tests/cpu/compare_with_acts_seeding.cpp

```diff
@@ -42,6 +41,57 @@
 // System include(s).
 #include <limits>
 
+// We need to define a 'SpacePointCollector' that will bridge
+// between the ACTS internal EDM and the experiment EDM
+// This class will accept a user-defined collection of space points and
+// need to implement some _impl function to instruct ACTS how to retrieve the
+// required quantities
+
+// The internal details of this class are totally up to the user
+class SpacePointCollector {
+    public:
+    friend Acts::SpacePointContainer<SpacePointCollector,
+                                     Acts::detail::RefHolder>;
+    using ValueType = SpacePoint;
+
+    explicit SpacePointCollector(
+        std::vector<const ValueType*>& externalCollection)
+        : m_storage(&externalCollection) {}
+
+    std::size_t size_impl() const { return storage().size(); }
+    float x_impl(std::size_t idx) const { return storage()[idx]->x(); }
+    float y_impl(std::size_t idx) const { return storage()[idx]->y(); }
+    float z_impl(std::size_t idx) const { return storage()[idx]->z(); }
+    float varianceR_impl(std::size_t idx) const {
+        return storage()[idx]->varianceR;
+    }
+    float varianceZ_impl(std::size_t idx) const {
+        return storage()[idx]->varianceZ;
+    }
```

Comment:
Interesting... 🤔 I **think** we should also be able to skip defining a `SpacePoint` class entirely, and just use `traccc::spacepoint` directly, no?

In any case, that's something that I'll be pretty interested to try for myself. As the spacepoints will soon need to go SoA as well anyway... 🤔

Reply:
indeed you can do it